common/src/graph.rs

```diff
@@ -231,27 +208,49 @@ impl Graph {
         NodeId(hash)
     }
 
-    /// Ensure all shorter neighbors of a not-yet-created child node exist and return them, excluding the given parent node
-    fn populate_shorter_neighbors_of_child(
+    /// Ensure all shorter neighbors of a not-yet-created neighbor node exist, and return them (including the given node)
+    /// in the form of ordered pairs containing the side they share with this not-yet-created neighbor node, and their node ID.
+    /// These ordered pairs will be sorted by side, based on enum order.
+    fn populate_shorter_neighbors_of_neighbor(
         &mut self,
-        parent: NodeId,
-        parent_side: Side,
+        node: NodeId,
+        side: Side,
     ) -> impl Iterator<Item = (Side, NodeId)> + Clone {
```

Comment:
Do we still need `+ Clone` on this?

Reply:
On second thought, do we need to return an iterator at *all*? It seems like we only care about the parent, so this could probably be simplified to just return `(Side, NodeId)` directly.

Reply:
We care about all elements of this iterator because of the
```rust
for (side, neighbor) in shorter_neighbors_of_neighbor
```
loop near the bottom of `insert_neighbor`. This is also why the iterator needs to be cloned, because it's first used to find the "parent", and then it is used to perform all the linking of neighbors.